arch/arm/core/aarch32/cortex_a_r/fault.c

```diff
@@ -191,6 +196,12 @@ bool z_arm_fault_undef_instruction(z_arch_esf_t *esf)
 		);
 #endif
 
+#if defined(CONFIG_GDBSTUB)
+	z_gdb_entry(esf, GDB_EXCEPTION_INVALID_INSTRUCTION);
+	/* Might not be fatal if GDB stub placed it in the code. */
+	return false;
```

Comment:
As I saw you `return false` after `z_gdb_entry`, would you like to elaborate on the details if there is a real fault instead of a gdb exception? You are going to send all faults to the gdb clients?